recipes/setup_delivery.rb

```diff
@@ -20,171 +20,5 @@
-  action :converge
-end
-
-# Set right permissions to delivery files
-machine_execute "Chown '/etc/delivery' to 'delivery' user" do
-  chef_server lazy { chef_server_config }
-  command 'chown -R delivery /etc/delivery'
-  machine delivery_server_hostname
-end
-
-machine_file 'delivery-server-cert' do
-  chef_server lazy { chef_server_config }
-  path lazy { "/var/opt/delivery/nginx/ca/#{delivery_server_fqdn}.crt" }
-  machine delivery_server_hostname
-  local_path lazy { "#{Chef::Config[:trusted_certs_dir]}/#{delivery_server_fqdn}.crt" }
-  action :download
-end
-
-# Create the default Delivery enterprise
-machine_execute 'Creating Enterprise' do
-  chef_server lazy { chef_server_config }
-  command lazy { delivery_enterprise_cmd }
-  machine delivery_server_hostname
-end
-
-# Download the credentials form the Delivery server
-machine_file "/tmp/#{node['delivery-cluster']['delivery']['enterprise']}.creds" do
-  chef_server lazy { chef_server_config }
-  machine delivery_server_hostname
-  local_path "#{cluster_data_dir}/#{node['delivery-cluster']['delivery']['enterprise']}.creds"
-  action :download
-end
-
-#########################################################################
-# Create Delivery builders
-#########################################################################
-
-# Create the Delivery builder role
-chef_role 'delivery_builders' do
-  chef_server lazy { chef_server_config }
-  description 'Base Role for the Delivery Build Nodes'
-  run_list builder_run_list
-end
-
-# Provision our builders in parallel
-machine_batch "#{node['delivery-cluster']['builders']['count']}-build-nodes" do
-  1.upto(node['delivery-cluster']['builders']['count']) do |i|
-    machine delivery_builder_hostname(i) do
-      chef_server lazy { chef_server_config }
-      role 'delivery_builders'
-      add_machine_options(
-        convergence_options: {
-          chef_config_text: "encrypted_data_bag_secret File.join(File.dirname(__FILE__), 'encrypted_data_bag_secret')",
-          ssl_verify_mode: :verify_none
-        }
-      )
-      provisioning.specific_machine_options('builders', i).each do |option|
-        add_machine_options option
-      end
-      files lazy {
-        {
-          "/etc/chef/trusted_certs/#{chef_server_fqdn}.crt" => "#{Chef::Config[:trusted_certs_dir]}/#{chef_server_fqdn}.crt",
-          "/etc/chef/trusted_certs/#{delivery_server_fqdn}.crt" => "#{Chef::Config[:trusted_certs_dir]}/#{delivery_server_fqdn}.crt",
-          '/etc/chef/encrypted_data_bag_secret' => "#{cluster_data_dir}/encrypted_data_bag_secret"
-        }
-      }
-      attributes lazy { builders_attributes }
-      converge true
-      action :converge
-    end
-  end
-end
-
-# Print the generated Delivery server credentials
-ruby_block 'print-delivery-credentials' do
-  block do
-    puts File.read("#{cluster_data_dir}/#{node['delivery-cluster']['delivery']['enterprise']}.creds")
-  end
-end
+include_recipe 'delivery-cluster::setup_delivery_server'
```

Comment:
I didn't know if we wanted to expose these as tasks in a Rakefile. Will do that.